[WIP] - Status message

superjax opened the pull request:
This follows #131.

This includes a status message per #132, and it also allows onboard commands to be accepted in the event of RC loss (for RTH functionality).  Still needs to be tested.

@dpkoch, I'm sorry, I had to mess up your beautiful `mux.c` functions.  If you have a better way to do this logic, then I'm all ears.  We now have 3 commands we have to mux, instead of 2.  Anyway.  I'd love your input.

superjax commented on src/mavlink_stream.c:
```diff
@@ -30,21 +67,14 @@ static void mavlink_send_heartbeat(void)
       control_mode = MODE_ROLLRATE_PITCHRATE_YAWRATE_THROTTLE;
-  }
 
   mavlink_msg_heartbeat_send(MAVLINK_COMM_0,
```
I don't love how much I'm abusing this message.  I considered creating a ROSFLIGHT_STATUS message which could be a lot easier to understand (as opposed to hijacking the default MAVLink stuff.)  What do you think?

superjax commented:
Flew on hardware, seems to be working properly.

dpkoch commented on src/mux.c:
```diff
@@ -137,60 +158,77 @@ static bool stick_deviated(mux_channel_t channel)
 {
-  bool rc_override;
+  bool rc_override = (rc_switch_mapped(RC_SWITCH_ATT_OVERRIDE) && rc_switch(RC_SWITCH_ATT_OVERRIDE))
+                      || stick_deviated(channel) || muxes[channel].rc->active;
```
Won't this always go into rc override if not in failsafe mode since rc->active is always true if not in failsafe mode and you've got an OR here? I was really trying to move way from these active fields since in my opinion they make the logic more confusing; the only reason I didn't delete it entirely was because I needed some way of knowing which values were valid in the offboard control message.

superjax commented:
I think I fixed the logic.

So, the reason I implemented this is because of #132.  I believe that there is a solid case for RTH on rc dropout.  Not completely sold, but I think that is the use case.
